src/app/components/MainLayout.tsx

```diff
@@ -0,0 +1,31 @@
+import Header from "@/app/components/Header";
+import React, {ReactNode} from 'react';
+import Sidebar from "@/app/components/Sidebar";
+import Footer from "@/app/components/Footer";
+
+interface LayoutProps {
+  children: ReactNode;
+}
+
+const MainLayout = ({children}: LayoutProps) => {
```

Comment:
You can write type PropsWithChildren, if you use only children, or if you want use children and another props you can use PropsWithChildren<interface> and then as example 
const {childre, otherprops} = props